dev/run-pip-tests

```diff
@@ -120,6 +122,10 @@ for python in "${PYTHON_EXECS[@]}"; do
     python "$FWDIR"/dev/pip-sanity-check.py
     echo "Run the tests for context.py"
     python "$FWDIR"/python/pyspark/context.py
+    if [ -n "$TEST_PYARROW" ]; then
+      echo "Run tests for pyarrow"
+      SPARK_TESTING=1 "$FWDIR"/bin/pyspark pyspark.sql.tests ArrowTests
```

Comment:
Yeah, this will definitely be removed at some point.  I was working with @holdenk to set this up as a temporary way to get the Python Arrow tests to run.  I'll look into using the pip packages and see if that can be used instead of this.